docs/how-tos/create-soap-endpoint.rst

```diff
@@ -0,0 +1,220 @@
+.. _create_soap_endpoint:
+
+Create a SOAP endpoint
+======================
+
+:author: `fjf2002 <https://github.com/fjf2002>`_
+
+PostgREST now has XML support. With a bit of work, SOAP endpoints become possible.
+
+Please note that PostgREST supports just ``text/xml`` MIME type in request/response headers ``Content-Type`` and ``Accept``.
+If you have to use other MIME types such as ``application/soap+xml``, you could manipulate the headers in your reverse proxy.
+
+
+
+Minimal Example
+---------------
+This example will simply return the request body, inside a tag ``therequestbodywas``.
+
+Add the following function to your PostgreSQL database:
+
+.. code-block:: postgres
+
+   CREATE OR REPLACE FUNCTION my_soap_endpoint(xml) RETURNS xml AS $$
+   DECLARE
+     nsarray CONSTANT text[][] := ARRAY[
+       ARRAY['soapenv', 'http://schemas.xmlsoap.org/soap/envelope/']
+     ];
+   BEGIN
+     RETURN xmlelement(
+       NAME "soapenv:Envelope",
+       XMLATTRIBUTES('http://schemas.xmlsoap.org/soap/envelope/' AS "xmlns:soapenv"),
+       xmlelement(NAME "soapenv:Header"),
+       xmlelement(
+         NAME "soapenv:Body",
+         xmlelement(
+           NAME theRequestBodyWas,
+           (xpath('/soapenv:Envelope/soapenv:Body', $1, nsarray))[1]
+         )
+       )
+    );
+   END;
+   $$ LANGUAGE plpgsql;
+
+Do not forget to refresh the :ref:`PostgREST schema cache <schema_reloading>`.
+
+Use ``curl`` for a first test:
+
+.. code-block:: bash
+
+   curl --data '
+     <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+       <soapenv:Header/>
+       <soapenv:Body>
+         <mySOAPContent>
+           My SOAP Content
+         </mySOAPContent>
+       </soapenv:Body>
+     </soapenv:Envelope>' \
+     --header 'Content-Type: text/xml' \
+     --header 'Accept: text/xml' \
+     http://localhost:3000/rpc/my_soap_endpoint
+
+The output should contain the original request body within the ``therequestbodywas`` entity,
+and should roughly look like:
+
+.. code-block:: xml
+
+    <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+      <soapenv:Header/>
+      <soapenv:Body>
+        <therequestbodywas>
+          <soapenv:Body xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+            <mySOAPContent>
+              My SOAP Content
+            </mySOAPContent>
+          </soapenv:Body>
+        </therequestbodywas>
+      </soapenv:Body>
+    </soapenv:Envelope>
+
+Unforunately the ``Accept: text/xml`` header is currently mandatory concerning PostgREST, otherwise it will respond
```

Comment:
[Spellcheck](https://github.com/PostgREST/postgrest-docs/runs/7086928620?check_suite_focus=true) did its job here:

```suggestion
Unfortunately the ``Accept: text/xml`` header is currently mandatory concerning PostgREST, otherwise it will respond
```